llama_index/agent/openai_agent.py

```diff
@@ -53,6 +55,38 @@ def call_function(
     )
 
 
+class StreamingChatResponse:
+    """Streaming chat response to user and writing to chat history."""
+
+    def __init__(self, chat_stream: ChatResponseGen) -> None:
+        self.chat_stream = chat_stream
+        self.queue: Queue = Queue()
```

Comment:
nit: may be useful to have an optional constructor param here for a user-provided `Queue` instance. So they can optionally provide one with its own `maxsize` settings.

Reply:
I'll see if there's a nice way to squeeze this param in somehow

Reply:
@sourabhdesai well tbh, the queue is kind of internal to each chat response. 

I don't see an advantage to providing your own queue instance, because then it would end up being shared between multiple chat responses (which sounds a little scary lol)

Was there a use-case you had in mind for this, other than setting max size?